### ISSUE TITLE
minor screen corruption

### ISSUE THREAD

**Issue by ThomasWaldmann** [QubesOS/qubes-issues#4602]
### Qubes OS version:

4.0.1-rc1 + updates

### Affected component(s): 

---

### Steps to reproduce the behavior:

I have a Thinkpad T470s with i7-6600u (integrated graphics).

I boot the machine, log in, work with it.

### Expected behavior:

No distortions.

### Actual behavior:

Sometimes I see screen corruption, usually the text inside an xfce "window button" (to the right of the qubes main menu) is distorted.

Interestingly, if i move the mouse cursor there, the distortion vanishes and comes back when i moved it away again.

### General notes:

Doesn't always happen. Did not see it yet before log in.

---

**Comment by DemiMarie** [comment]
I can reproduce this.  It happens regularly on my system.

**Comment by DemiMarie** [comment]
@marmarek I believe that Qubes has not always had this problem on my system, but I will try your suggestion.  Am I correct that the kernel-latest package is fully supported by the Qubes team, just less tested?

**Comment by marmarek** [comment]
>  Am I correct that the kernel-latest package is fully supported by the Qubes team, just less tested?

Yes.

**Comment by DemiMarie** [comment]
I just booted with the `kernel-latest` package.  No corruption so far.

**Comment by ThomasWaldmann** [comment]
@DemiMarie did you see it that frequently? I've seen it now and then, but guess I would take some weeks to be sure it is gone.

**Comment by DemiMarie** [comment]
@ThomasWaldmann I *always* saw it.

**Comment by SvenSemmler** [comment]
I have the same observation as @DemiMarie. My Thinkpad P51 has both NVIDIA and Intel graphics. I use the Intel graphics for better battery life. My system started showing distortions a few weeks ago after an update. When I switch to NVIDIA graphics (in the BIOS) they don't happen. When they happen (with Intel graphics) I can make them go away by switching to another desktop and back or by opening another window and closing it (i3 wm ... basically forcing a redraw).
I figured it's a bug in one of the updates regarding the Intel graphics and many people would see it / it would be fixed upstream sooner than later. Since I can switch between NVIDIA and Intel and it only happens with Intel my assumption is that the kernel driver is at fault?

**Comment by SvenSemmler** [comment]
Yup. Still happens, but less often and weirdly self correcting (corruption is there for a moment and then it's gone). It looks as if one has wiped across the screen or maybe imagine an old analog TV that has bad reception. @DemiMarie, @ThomasWaldmann is that what you are seeing?

**Comment by ThomasWaldmann** [comment]
@SvenSemmler i am seeing only the stuff i described in the top post, no more, no less.

**Comment by SvenSemmler** [comment]
Thanks @ThomasWaldmann . I think we might talk about two different issues then. The corruption you are seeing is in dom0 while the corruptions I am seeing happen within windows (e.g. firefox content) of fedora-based qubes but not in Windows based qubes. I'll do some more experiments (with debian template and maybe older fedora templates) and then create another ticket.

**Comment by jpouellet** [comment]
I've seen occasional corruption as well, limited so far to:
- corrupted text in the top-panel xfce window switcher (bars through the top and bottom of words, or something like that)
- blank and/or unaligned workspaces in the xfce workspace-switcher in the top panel
- blank xfce tray icons
- big black squares/rectangles of the screen not apparently correlating with any particular window (which do not show up in screenshots, so I just assumed hardware problems)
